deployment/ccip/changeset/testhelpers/test_helpers.go

```diff
@@ -793,6 +809,17 @@ func ConvertSolanaCrossChainAmountToBigInt(amount ccip_router.CrossChainAmount)
 	return big.NewInt(0).SetBytes(bytes)
 }
 
+func InferTokenProgramID(ctx context.Context, client *rpc.Client, tokenPubKey solana.PublicKey, fallbackTokenProgramID solana.PublicKey) (solana.PublicKey, error) {
+	tokenAcctInfo, err := client.GetAccountInfo(ctx, tokenPubKey)
+	if errors.Is(err, rpc.ErrNotFound) {
+		return fallbackTokenProgramID, nil
```

Comment:
You should inline the fallback and assume that if the account doesn't exist it'll get created by the associated token program (which will use token-2022)